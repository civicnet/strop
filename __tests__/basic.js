'use strict'

const StrOP = require('..');

it('exists', () => {
    expect(StrOP).toBeInstanceOf(Function);
});

it('is unchanged', () => {
    expect(Object.getOwnPropertyNames(StrOP).sort()).toMatchSnapshot();
});

describe('instance', () => {
    it('requires a name', () => {
        expect(() => new StrOP).toThrowError(TypeError);
    });

    it('constructs', () => {
        expect(() => new StrOP('Basic')).not.toThrow();
    });

    it('is a function', () => {
        expect(new StrOP('Basic')).toBeInstanceOf(Function);
    });

    it('has the correct name', () => {
        expect(new StrOP('Basic').name).toEqual('Basic');
    });

    it('is not a constructor', () => {
        expect(() => new (new StrOP('Basic'))).toThrowError(TypeError);
    });

    it('is unchanged', () => {
        expect(Object.getOwnPropertyNames(new StrOP('Basic')).sort()).toMatchSnapshot();
    });

    it('tags strings', () => {
        const test = new StrOP('Basic');

        expect(test`Some string`).toBeDefined();
    });

    it('expects valid indentation characters', () => {
        const test = new StrOP('Basic');

        expect(() => { test.indent = null; }).toThrowError(TypeError);
        expect(() => { test.indent = 7; }).toThrowError(TypeError);
        expect(() => { test.indent = ''; }).toThrowError(TypeError);
        expect(() => { delete test.indent; }).toThrowError(TypeError);
    });

    it('freezes results', () => {
        const test = new StrOP('Basic');

        expect(Object.isFrozen(test`Some string`)).toBeTruthy();
    });
});
