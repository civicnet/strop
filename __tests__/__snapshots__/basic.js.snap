// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`instance is unchanged 1`] = `
Array [
  "length",
  "name",
  "arguments",
  "caller",
  "prototype",
  "indent",
]
`;

exports[`is unchanged 1`] = `
Array [
  "length",
  "prototype",
  "name",
]
`;
